Description: Make log files readable to group "adm", if present.
Author: Martin Pitt <mpitt@debian.org>
Bug-Ubuntu: https://launchpad.net/bugs/345953

--- a/scheduler/log.c
+++ b/scheduler/log.c
@@ -32,6 +32,7 @@
 #include "cupsd.h"
 #include <stdarg.h>
 #include <syslog.h>
+#include <grp.h>
 
 
 /*
@@ -61,6 +62,7 @@
 		filename[1024],		/* Formatted log filename */
 		*ptr;			/* Pointer into filename */
   const char	*logptr;		/* Pointer into log filename */
+  struct group  *loggrp;		/* Group entry of log filename */
 
 
  /*
@@ -71,6 +73,11 @@
     return (1);
 
  /*
+  * Use adm group if possible, fall back to Group
+  */
+ loggrp = getgrnam("adm");
+
+ /*
   * Format the filename as needed...
   */
 
@@ -180,7 +187,7 @@
       * Change ownership and permissions of non-device logs...
       */
 
-      fchown(cupsFileNumber(*lf), RunUser, Group);
+      fchown(cupsFileNumber(*lf), RunUser, loggrp ? loggrp->gr_gid : Group);
       fchmod(cupsFileNumber(*lf), LogFilePerm);
     }
   }
@@ -219,7 +226,7 @@
     * Change ownership and permissions of non-device logs...
     */
 
-    fchown(cupsFileNumber(*lf), RunUser, Group);
+    fchown(cupsFileNumber(*lf), RunUser, loggrp ? loggrp->gr_gid : Group);
     fchmod(cupsFileNumber(*lf), LogFilePerm);
   }
 
